### PR TITLE
SJQ2T-24 and SJQ2T-41: added registration db script, entities, dtos, and repositories

### PR DESCRIPTION
**Pull request by ivanfhtec**
also switched repos to use optional, where applicable
removed Utils as we'll be using UUIDs instead of generating our own strings

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/SkyNest/controller/UserController.java [hunk 1]
```diff
@@ -18,12 +17,10 @@ public class UserController {
   @PostMapping("/register")
   public UserResponse registerUser(@RequestBody UserRegisterRequest userRegisterRequest) {
 
-    ModelMapper modelMapper = new ModelMapper();
+    UserDto userDto =
+        userService.registerUser(new ModelMapper().map(userRegisterRequest, UserDto.class));
```
Comment (markobarackovhtec) [inline backend/src/main/java/com/htecgroup/SkyNest/controller/UserController.java]: Can we maybe create model mapper as bean in configuration, and to autowire that where we need it ?

File: backend/src/main/java/com/htecgroup/SkyNest/model/enitity/RoleEntity.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@AllArgsConstructor
+public class RoleEntity implements Serializable {
+
+  private static final long serialVersionUID = -6954026563601179057L;
+
+  public static final String ROLE_WORKER = "role_worker";
+  public static final String ROLE_MANAGER = "role_manager";
+  public static final String ROLE_ADMIN = "role_admin";
+
+  @Id @GeneratedValue private UUID id;
```
Comment (markobarackovhtec) [inline backend/src/main/java/com/htecgroup/SkyNest/model/enitity/RoleEntity.java]: I see that somewhere in a code you use annotation above some code, and somewhere inline, maybe is better to be consistent about this and for example to use it always above

File: backend/src/main/java/com/htecgroup/SkyNest/repository/RoleRepository.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.htecgroup.SkyNest.model.enitity.RoleEntity;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Repository
+public interface RoleRepository extends CrudRepository<RoleEntity, UUID> {
```
Comment (markobarackovhtec) [inline backend/src/main/java/com/htecgroup/SkyNest/repository/RoleRepository.java]: why somewhere CrudRepository and somewhere JpaRepository ?

File: backend/src/main/resources/db/migration/V2__registration_entities.sql [hunk 4]
```diff
@@ -0,0 +1,69 @@
+--  modify old user entity
+ALTER TABLE `user`
```
Comment (markobarackovhtec) [inline backend/src/main/resources/db/migration/V2__registration_entities.sql]: My suggestion here is to use multiple script for every alter, every create, every insert...

File: backend/src/main/java/com/htecgroup/SkyNest/controller/UserController.java (hunk 1 above)
Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/SkyNest/controller/UserController.java]: Ok, I've added a bean to SkyNestApplication.java and autowired it where necessary.

File: backend/src/main/java/com/htecgroup/SkyNest/model/enitity/RoleEntity.java (hunk 2 above)
Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/SkyNest/model/enitity/RoleEntity.java]: We're using Google's formatter, looks like it tries to fit all annotations in one line.

File: backend/src/main/java/com/htecgroup/SkyNest/repository/RoleRepository.java (hunk 3 above)
Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/SkyNest/repository/RoleRepository.java]: JpaRepository extends CrudRepository, PagingAndSortingRepository, QueryByExampleExecutor, and Repository.

I thought that for roles, we don't need most of these. Should we have every repository extend JpaRepository?

File: backend/src/main/resources/db/migration/V2__registration_entities.sql (hunk 4 above)
Comment (ivanfhtec) [reply backend/src/main/resources/db/migration/V2__registration_entities.sql]: Is it enough to break the whole script into:
- modify user entity
- create role
- create company
- add foreign keys
- init role and company

In other words, break it up where I've left comments.

Or should I break it down further?

File: backend/src/main/java/com/htecgroup/SkyNest/repository/RoleRepository.java (hunk 3 above)
Comment (markobarackovhtec) [reply backend/src/main/java/com/htecgroup/SkyNest/repository/RoleRepository.java]: Valid answer, if you are sure that you are not going to use the functions provided by JpaRepository, use CrudRepository :)

File: backend/src/main/java/com/htecgroup/SkyNest/model/enitity/RoleEntity.java (hunk 2 above)
Comment (markobarackovhtec) [reply backend/src/main/java/com/htecgroup/SkyNest/model/enitity/RoleEntity.java]: Then im ok with it

File: backend/src/main/resources/db/migration/V2__registration_entities.sql (hunk 4 above)
Comment (markobarackovhtec) [reply backend/src/main/resources/db/migration/V2__registration_entities.sql]: I would say script per alter, script per create, scrip per insert (you can use one insert for multiple values in same table)

Comment (ivanfhtec) [reply backend/src/main/resources/db/migration/V2__registration_entities.sql]: Ok, I'll make one script per statement.

Comment (markobarackovhtec) [reply backend/src/main/resources/db/migration/V2__registration_entities.sql]: In that case, scripts will have understandable names and we will know which script does what just by reading the name

**Comment by markobarackovhtec** [comment]
Gj for using optional with suppliers.
Only thing that i dont see are unit tests :)

**Comment by ivanfhtec** [comment]
I've added some tests for UserServiceImpl.

**Review by markobarackovhtec — APPROVED** [review]

**Comment by markobarackovhtec** [comment]
Now i like it more :)

**Pull request merged**